cassandra/src/main/scala/geotrellis/spark/io/cassandra/CassandraRDDReader.scala

```diff
@@ -0,0 +1,66 @@
+package geotrellis.spark.io.cassandra
+
+import geotrellis.spark.io.avro.codecs.KeyValueRecordCodec
+import geotrellis.spark.util.KryoWrapper
+import geotrellis.spark.{Boundable, KeyBounds}
+import geotrellis.spark.io.avro.{AvroEncoder, AvroRecordCodec}
+import geotrellis.spark.io.index.{IndexRanges, MergeQueue}
+import org.apache.avro.Schema
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import scala.collection.JavaConversions._
+import scala.reflect.ClassTag
+
+object CassandraRDDReader {
+  def read[K: Boundable : AvroRecordCodec : ClassTag, V: AvroRecordCodec : ClassTag](
+    table: String,
+    queryKeyBounds: Seq[KeyBounds[K]],
+    decomposeBounds: KeyBounds[K] => Seq[(Long, Long)],
+    filterIndexOnly: Boolean,
+    writerSchema: Option[Schema] = None,
+    numPartitions: Option[Int] = None
+  )(implicit sc: SparkContext, instance: CassandraInstance): RDD[(K, V)] = {
+    if (queryKeyBounds.isEmpty) return sc.emptyRDD[(K, V)]
+
+    val includeKey = (key: K) => queryKeyBounds.includeKey(key)
+    val _recordCodec = KeyValueRecordCodec[K, V]
+    val kwWriterSchema = KryoWrapper(writerSchema) //Avro Schema is not Serializable
+
+    val ranges = if (queryKeyBounds.length > 1)
+        MergeQueue(queryKeyBounds.flatMap(decomposeBounds))
+      else
+        queryKeyBounds.flatMap(decomposeBounds)
+
+    val bins = IndexRanges.bin(ranges, numPartitions.getOrElse(sc.defaultParallelism))
+
+    val rdd =
+      sc.parallelize(bins, bins.size)
+        .mapPartitions { partition: Iterator[Seq[(Long, Long)]] =>
+          val session = instance.session
+
+          val statement = session.prepare(s"SELECT value FROM ${instance.keySpace}.${table} WHERE key = ?")
```

Comment:
probably that would be faster to use range queries there, but not sure (the only way to figure it out -- to try)